src/gdb_main.c

```diff
@@ -210,52 +204,10 @@ int gdb_main_loop(target_controller_s *tc, bool in_syscall)
 			 * This packet isn't documented as being mandatory,
 			 * but GDB doesn't work without it.
 			 */
-
-			if (!cur_target) {
-				/* Report "target exited" if no target */
-				gdb_putpacketz("W00");
-				break;
-			}
-
-			/* Wait for target halt */
-			target_addr_t watch;
-			target_halt_reason_e reason;
-			while (true) {
-				reason = target_halt_poll(cur_target, &watch);
-				if (reason != TARGET_HALT_RUNNING)
-					break;
-
-				const char gdb_ch = (char)gdb_if_getchar_to(0);
-				if (gdb_ch == '\x03' || gdb_ch == '\x04')
-					target_halt_request(cur_target);
-
-				platform_pace_poll();
-
-#ifdef ENABLE_RTT
-				if (rtt_enabled)
-					poll_rtt(cur_target);
-#endif
-			}
-			SET_RUN_STATE(false);
-
-			/* Translate reason to GDB signal */
-			switch (reason) {
-			case TARGET_HALT_ERROR:
-				gdb_putpacket_f("X%02X", GDB_SIGLOST);
-				morse("TARGET LOST.", true);
-				break;
-			case TARGET_HALT_REQUEST:
-				gdb_putpacket_f("T%02X", GDB_SIGINT);
-				break;
-			case TARGET_HALT_WATCHPOINT:
-				gdb_putpacket_f("T%02Xwatch:%08X;", GDB_SIGTRAP, watch);
-				break;
-			case TARGET_HALT_FAULT:
-				gdb_putpacket_f("T%02X", GDB_SIGSEGV);
-				break;
-			default:
-				gdb_putpacket_f("T%02X", GDB_SIGTRAP);
-			}
+			if (cur_target)
```

Comment:
Does this not need to call the new gdb_poll_target() before sending back a `W00` or no-response? If not, please add documentation as to why this otherwise disjointed-seeming approach works and what the new logic is